internal/pkg/config/config.go

```diff
@@ -0,0 +1,83 @@
+package config
+
+import (
+	"log/slog"
+	"net"
+	"os"
+	"runtime/debug"
+	"time"
+
+	"github.com/kelseyhightower/envconfig"
+	"github.com/stackus/dotenv"
+)
+
+type HTTP struct {
+	Host              string        `default:"0.0.0.0"`
+	Port              string        `default:"8080"`
+	ReadHeaderTimeout time.Duration `envconfig:"HTTP_READ_HEADER_TIMEOUT" default:"1s"`
+}
+
+func (h HTTP) Address() string {
+	return net.JoinHostPort(h.Host, h.Port)
+}
+
+func (h HTTP) LogValue() slog.Value {
+	return slog.GroupValue(
+		slog.String("host", h.Host),
+		slog.String("port", h.Port),
+		slog.String("readHeaderTimeout", h.ReadHeaderTimeout.String()),
+	)
+}
+
+type VCSInfo struct {
+	Revision string
+	Tag      string
+	Time     string
+}
+
+// Config is the type that holds the configuration for the service.
+type Config struct {
+	VCS             VCSInfo
+	ServiceName     string `envconfig:"SERVICE_NAME" default:"noname"`
+	Environment     string `envconfig:"ENVIRONMENT" default:"dev"`
+	LogLevel        string `envconfig:"LOG_LEVEL" default:"DEBUG"`
```

Comment:
Are we store config values into .env file? i heard something that storing it into .toml file better approach

Reply:
The configuration has always been and will continue to be performed using environment variables because some variables may vary within the same deployment environment. For example, for AB tests, we can use different configurations in a single deployment for several instances of our application. 
At this stage of development, how to store the config values doesn't matter much, as it will heavily depend on the specific deployment technology.

For now, populating from a dotenv file is the most straightforward and out-of-the-box solution. May be in the future it can vary.